apps/web/src/actions/sdk/runDocumentAction.ts

```diff
@@ -22,26 +22,30 @@ export async function runDocumentAction({
   const result = await getLatitudeApiKey()
   if (result.error) return result
 
-  const stream = createStreamableValue()
-  const latitudeApiKey = result.value.token
+  const stream = createStreamableValue<ChainEvent, Error>()
 
-  const sdk = new LatitudeSdk({ latitudeApiKey, projectId })
-  await sdk.runDocument({
-    params: { commitUuid, documentPath },
-    onMessage: (message) => {
-      stream.update({
-        event: message.event,
-        data: message.data,
-      })
-    },
-    onError: (error) => {
-      stream.error({
-        name: error.name,
-        message: error.message,
-        stack: error.stack,
-      })
-    },
-  })
+  const runSdk = async () => {
+    const sdk = new LatitudeSdk({
+      latitudeApiKey: result.value.token,
+      projectId,
+    })
+    sdk.runDocument({
+      params: { commitUuid, documentPath },
+      onMessage: (message) => {
+        stream.update(message)
+      },
+      onError: (error) => {
+        stream.error({
+          name: error.name,
+          message: error.message,
+          stack: error.stack,
+        })
+      },
+      onFinished: () => stream.done(),
+    })
+  }
+
+  runSdk()
```

Comment:
Why is this defined as a function? Can't you just do it directly?

Reply:
Removed